subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/AbstractValidatingProperty.java

```diff
@@ -50,16 +55,27 @@ public static void reportValueNotSet(String propertyName, TypeValidationContext
 
     @Override
     public void validate(PropertyValidationContext context) {
-        Object unpacked = DeferredUtil.unpackOrNull(value.call());
```

Comment:
`unpackOrNull` unpacked callables and then resolved deferreds (provider, factory). We now do it in two steps - we only do the second step if a validation action requires so.